x-pack/plugins/actions/server/builtin_action_types/lib/get_custom_agents.ts

```diff
@@ -22,14 +22,36 @@ export function getCustomAgents(
   logger: Logger,
   url: string
 ): GetCustomAgentsResponse {
-  const proxySettings = configurationUtilities.getProxySettings();
   const defaultAgents = {
     httpAgent: undefined,
     httpsAgent: new HttpsAgent({
       rejectUnauthorized: configurationUtilities.isRejectUnauthorizedCertificatesEnabled(),
     }),
   };
 
+  const proxySettings = configurationUtilities.getProxySettings();
+  const customHostSettings = configurationUtilities.getCustomHostSettings(url);
+  if (!proxySettings && !customHostSettings) {
+    return defaultAgents;
+  }
+
+  // update the defaultAgents.httpsAgent if configured
+  const tlsSettings = customHostSettings?.tls;
+  let agentOptions: AgentOptions | undefined;
+  if (tlsSettings) {
+    logger.debug(`Creating customized connection settings for: ${url}`);
+    agentOptions = defaultAgents.httpsAgent.options;
+
+    if (tlsSettings.certificateAuthoritiesData) {
+      agentOptions.ca = tlsSettings.certificateAuthoritiesData;
+    }
+
+    // see: src/core/server/elasticsearch/legacy/elasticsearch_client_config.ts
+    if (tlsSettings.rejectUnauthorized !== undefined) {
+      agentOptions.rejectUnauthorized = tlsSettings.rejectUnauthorized;
+    }
```

Comment:
Nit: you mentioned it in the asciidocs, but it's not abundantly clear here that this will inherit from `xpack.actions.rejectUnauthorized` (which defaults to `true`)

I would suggest just adding a comment:
```suggestion
    // Note: if custom TLS settings are defined but rejectUnauthorized is not set, fall back to the config for `xpack.actions.rejectUnauthorized` (default: true)
    // See also: src/core/server/elasticsearch/legacy/elasticsearch_client_config.ts
    if (tlsSettings.rejectUnauthorized !== undefined) {
      agentOptions.rejectUnauthorized = tlsSettings.rejectUnauthorized;
    }
```

Reply:
Added a number of comments in this file to hopefully clarify things.